include/librealsense2/dds/dds-device-impl.h

```diff
@@ -88,54 +93,89 @@ class dds_device::impl
-        _topic
-            = DDS_API_CALL( _participant->get()->create_topic( topic_name, topic_type->getName(), TOPIC_QOS_DEFAULT ) );
-
-        // CREATE THE READER
+        // Create the reader
         DataReaderQos rqos = DATAREADER_QOS_DEFAULT;
         rqos.data_sharing().off();
-        rqos.reliability().kind = RELIABLE_RELIABILITY_QOS;
-        rqos.durability().kind = VOLATILE_DURABILITY_QOS;
-        rqos.history().kind = KEEP_LAST_HISTORY_QOS;
-        rqos.history().depth = 10;
+        rqos.reliability().kind               = RELIABLE_RELIABILITY_QOS;
+        rqos.durability().kind                = VOLATILE_DURABILITY_QOS;
+        rqos.history().kind                   = KEEP_LAST_HISTORY_QOS;
+        rqos.history().depth                  = 10;
         rqos.endpoint().history_memory_policy = eprosima::fastrtps::rtps::PREALLOCATED_WITH_REALLOC_MEMORY_MODE;
-        _reader = DDS_API_CALL( _subscriber->create_datareader( _topic, rqos, nullptr ) );
-        if( _reader == nullptr )
+
+        _notifications_reader = DDS_API_CALL( _subscriber->create_datareader( _notifications_topic, rqos, nullptr ) );
+
+        LOG_DEBUG( "topic '" << topic_name << "' and reader created" );
+    }
+
+    void create_control_writer()
+    {
+        using namespace eprosima::fastdds::dds;
+
+        if (_publisher == nullptr)
         {
-            throw std::runtime_error( "Error creating a DDS reader for " + topic_name );
+            _publisher = DDS_API_CALL( _participant->get()->create_publisher( PUBLISHER_QOS_DEFAULT, nullptr ) );
         }
 
-        LOG_DEBUG( "topic '" << topic_name << "' created" );
+        // Create and register topic type
+        eprosima::fastdds::dds::TypeSupport topic_type( new librealsense::dds::topics::device::control::type );
+        DDS_API_CALL( topic_type.register_type( _participant->get() ) );
+
+        // Create the topic
+        std::string topic_name = librealsense::dds::topics::device::control::construct_topic_name( _info.topic_root );
+        //TODO - When the last dds_device destruct we should delete the topic
+        eprosima::fastdds::dds::Topic* _control_topic = DDS_API_CALL( _participant->get()->create_topic( topic_name,
+                                                                                                         topic_type->getName(),
+                                                                                                         TOPIC_QOS_DEFAULT ) );
+
+        // Create the writer
+        DataWriterQos wqos = DATAWRITER_QOS_DEFAULT;
+        wqos.data_sharing().off();
+        wqos.reliability().kind               = RELIABLE_RELIABILITY_QOS;
+        wqos.durability().kind                = VOLATILE_DURABILITY_QOS;
+        wqos.history().kind                   = KEEP_LAST_HISTORY_QOS;
+        wqos.history().depth                  = 10;
+        wqos.endpoint().history_memory_policy = eprosima::fastrtps::rtps::PREALLOCATED_WITH_REALLOC_MEMORY_MODE;
```

Comment:
Explain?

Reply:
PREALLOCATED_WITH_REALLOC_MEMORY_MODE is a history management mode that allocates some default memory size for requested history depth at initialization, but allows reallocating if history needs to grow.
This allows flexibility but still limits number of reallocations.
See [discussion](https://github.com/eProsima/Fast-DDS/discussions/2707)

Reply:
The code should explain why

Reply:
Done